### PR TITLE
refactor: use rpc_holder to receive RPC in pegasus_server_impl

### PR DESCRIPTION
**Pull request by Smityz**
### What problem does this PR solve? 
This PR plans to replace `rpc_replier` to `rpc_holder` in `pegasus_server_impl`

### What is changed and how it works?
just a refactor

### CODE REVIEW

File: src/server/pegasus_read_service.h [hunk 1]
```diff
@@ -6,6 +6,9 @@
 
 namespace pegasus {
 namespace server {
+
+typedef dsn::rpc_holder<::dsn::blob, ::dsn::apps::read_response> on_get_rpc;
```
Comment (neverchanje) [inline src/server/pegasus_read_service.h:10]:
```suggestion
typedef dsn::rpc_holder<::dsn::blob, ::dsn::apps::read_response> get_rpc;
```

File: src/server/pegasus_read_service.h [hunk 2]
```diff
@@ -23,8 +26,7 @@ class pegasus_read_service : public dsn::replication::replication_app_base,
 protected:
     // all service handlers to be implemented further
     // RPC_RRDB_RRDB_GET
-    virtual void on_get(const ::dsn::blob &args,
-                        ::dsn::rpc_replier<dsn::apps::read_response> &reply) = 0;
+    virtual void on_get(on_get_rpc rpc) = 0;
     // RPC_RRDB_RRDB_MULTI_GET
     virtual void on_multi_get(const dsn::apps::multi_get_request &args,
```
Comment (levy5307) [inline src/server/pegasus_read_service.h:31]: you can refactor on_multi_get(on_sort_count、on_ttl, and so on) also, by using rpc_holder

File: src/server/pegasus_read_service.h [hunk 3]
```diff
@@ -6,6 +6,14 @@
 
 namespace pegasus {
 namespace server {
+
+typedef dsn::rpc_holder<::dsn::blob, ::dsn::apps::read_response> get_rpc;
+typedef dsn::rpc_holder<dsn::apps::multi_get_request, dsn::apps::multi_get_response> multi_get_rpc;
+typedef dsn::rpc_holder<::dsn::blob, dsn::apps::count_response> sortkey_count_rpc;
+typedef dsn::rpc_holder<::dsn::blob, dsn::apps::ttl_response> ttl_rpc;
+typedef dsn::rpc_holder<::dsn::apps::get_scanner_request, dsn::apps::scan_response> get_scanner_rpc;
+typedef dsn::rpc_holder<::dsn::apps::scan_request, dsn::apps::scan_response> scan_rpc;
```
Comment (hycdong) [inline src/server/pegasus_read_service.h:15]: I suggest remove "::" before `dsn`.

File: src/server/pegasus_server_impl.cpp [hunk 4]
```diff
@@ -237,14 +237,14 @@ int pegasus_server_impl::on_batched_write_requests(int64_t decree,
...
-                                 ::dsn::rpc_replier<::dsn::apps::read_response> &reply)
+void pegasus_server_impl::on_get(get_rpc rpc)
 {
+    const ::dsn::blob &key = rpc.request();
     dassert(_is_open, "");
     _pfc_get_qps->increment();
     uint64_t start_time = dsn_now_ns();
 
-    ::dsn::apps::read_response resp;
+    ::dsn::apps::read_response &resp = rpc.response();
```
Comment (levy5307) [inline src/server/pegasus_server_impl.cpp:247]: auto

File: src/server/pegasus_server_impl.cpp [hunk 5]
```diff
@@ -237,14 +237,14 @@ int pegasus_server_impl::on_batched_write_requests(int64_t decree,
     return _server_write->on_batched_write_requests(requests, count, decree, timestamp);
 }
 
-void pegasus_server_impl::on_get(const ::dsn::blob &key,
-                                 ::dsn::rpc_replier<::dsn::apps::read_response> &reply)
+void pegasus_server_impl::on_get(get_rpc rpc)
 {
+    const ::dsn::blob &key = rpc.request();
```
Comment (levy5307) [inline src/server/pegasus_server_impl.cpp:242]: const auto&

File: src/server/pegasus_server_impl.cpp [hunk 6]
```diff
@@ -719,19 +716,17 @@ void pegasus_server_impl::on_multi_get(const ::dsn::apps::multi_get_request &req
...
+void pegasus_server_impl::on_sortkey_count(sortkey_count_rpc rpc)
 {
     dassert(_is_open, "");
 
     _pfc_scan_qps->increment();
     uint64_t start_time = dsn_now_ns();
 
-    ::dsn::apps::count_response resp;
+    const auto &hash_key = rpc.request();
+    auto resp = rpc.response();
```
Comment (hycdong) [inline src/server/pegasus_server_impl.cpp:729]: auto &resp

**Pull request merged**